lnwallet/channel_test.go

```diff
@@ -8028,12 +8064,21 @@ func TestChannelMaxFeeRate(t *testing.T) {
 	}
 	defer cleanUp()
 
-	assertMaxFeeRate(aliceChannel, 1.0, 0, 676794154)
-	assertMaxFeeRate(aliceChannel, 0.001, 0, 676794)
-	assertMaxFeeRate(aliceChannel, 0.000001, 0, 676)
-	assertMaxFeeRate(aliceChannel, 0.0000001, 0, chainfee.FeePerKwFloor)
+	// If the maximum fee is lower than the network fee and above the
+	// min relay fee, use the maximum fee
+	assertIdealFeeRate(aliceChannel, 700000000, 0, 0, 1.0, 676794154)
 
-	// Check that anchor channels are capped at their max fee rate.
+	// If the network fee is lower than the max fee and above the min
+	// relay fee, use the network fee.
+	assertIdealFeeRate(aliceChannel, 500000000, 0, 0, 1.0, 500000000)
+
+	// If the fee is below the minimum relay fee, and the min relay fee
+	// is less than our max fee, use the minimum relay fee.
+	assertIdealFeeRate(
+		aliceChannel, 500000000, 600000000, 0, 1.0, 600000000,
+	)
+
+	// Test ideal fee rates for an anchor channel
```

Comment:
Could make this into a sub-test: https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks

Reply:
I don't know that we can combine the logic here with `assertIdealFeeRate` as there are differing values which is one of the main advantages of subtests.  I think `CreateTestChannels` could be combined, but not sure it has a true benefit.